packages/catalog-process/src/services/catalogService.ts

```diff
@@ -3083,14 +3088,133 @@ export function catalogServiceBuilder(
           eserviceInstanceInterfaceData,
           config.eserviceTemplateDocumentsContainer,
           fileManager,
-          readModelService,
           ctx
         );
 
       await repository.createEvent(addDocumentEvent);
 
       return updatedEService;
     },
+    async createInstanceDescriptor(
+      eserviceId: EServiceId,
+      eserviceInstanceDescriptorSeed: catalogApi.EServiceInstanceDescriptorSeed,
+      ctx: WithLogger<AppContext>
+    ): Promise<Descriptor> {
+      ctx.logger.info(
+        `Creating Instance Descriptor for EService ${eserviceId}`
+      );
+
+      const eservice = await retrieveEService(eserviceId, readModelService);
+
+      await assertRequesterIsDelegateProducerOrProducer(
+        eservice.data.producerId,
+        eservice.data.id,
+        ctx.authData,
+        readModelService
+      );
+
+      assertHasNoDraftOrWaitingForApprovalDescriptor(eservice.data);
+      assertEServiceIsTemplateInstance(eservice.data);
+
+      const template = await retrieveEServiceTemplate(
+        eservice.data.templateRef.id,
+        readModelService
+      );
+
+      const latestDescriptor = getLatestDescriptor(eservice.data);
+      const templateVersion = template.versions.find(
+        (v) => v.id === latestDescriptor?.templateVersionRef?.id
+      );
+
+      if (!templateVersion) {
+        throw templateVersionNotFoundInEService(
```

Comment:
The name is misleading, it says that you did not find a template version in the e-service but actually, the error happens when the template version from the descriptor cannot be found in the e-service template

```suggestion
        throw descriptorTemplateVersionNotFound(
```